Automation to lay down F5 config for UPI deployment

bmarlow opened the pull request:
Automation will create:
TCP based VIPs for API and APP endpoints

API VIP is 6443
APP VIP is 80 and 443

Respective pools and node members are configurable in size by adding more nodes into the appropriate dictionaries.

bmarlow committed:
first pass

bmarlow committed:
general clean up

ddreggors commented:
I see the new role here but I do not see any code changes to use this role. 

How can we test this role and verify it if nothing calls it in the playbooks?
Can we rename this role to something that says F5 BigIP?

Generally, we would make use of a `when` clause in the playbook when importing roles like this that allows us to pass a variable (often boolean) that decides whether we use this role.

bmarlow committed:
Refactor so F5 config runs during common role

bmarlow commented:
Fair points, thanks for the direction.

I've updated it so that it whenever the common tasks are run, if the F5 var is enabled it will run the F5 config.

It has also been refactored so that it uses the existing role variables wherever possible.

bmarlow committed:
convert to role, make reachable by main playbooks

bmarlow committed:
add f5 config to dhcp playbooks

bmarlow committed:
add API int

bmarlow committed:
fix typos, correct pathing

bmarlow committed:
fix pathing, again.

bmarlow committed:
fix port numbers for api-int

bmarlow committed:
add persistence to VIPs

bmarlow committed:
add server and client ssl profiles update rollback

bmarlow committed:
update top level playbooks

bmarlow committed:
update for variable inheritance

bmarlow committed:
fix variable, but for real this time

bmarlow committed:
do kubernetes they said.  it'll be fun they said.

ddreggors committed:
Small changes

Change order of operations
Add better labels to some loops
Change pool member names
Add ICMP (node Alive) health checks to nodes

ddreggors committed:
Update bootstrap removal code

Use full name in name field
Add address as required
Give better loop labels

ddreggors committed:
More changes

Force download/unpack of oc binaries
Add port to pool member removal
Add retry logic to finish install

ddreggors committed:
Do not remove nodes 

Removing pool members also deletes the node. We do not want to delete 
nodes when pools are still using them. So we set preserve_node.

ddreggors commented:
```
PLAY RECAP ************************************************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=105  changed=47   unreachable=0    failed=0    skipped=30   rescued=0    ignored=0 
```

The latest changes were tested, and the cluster was built with the F5 role used. Cluster and F5 changes all checked out.
